test/integration/test_load_and_run_checkpoint.py

```diff
@@ -0,0 +1,160 @@
+    # model card: https://huggingface.co/torchao-testing/single-linear-Float8DynamicActivationFloat8WeightConfig-v2-0.13-dev
+    "torchao-testing/single-linear-Float8DynamicActivationFloat8WeightConfig-v2-0.13-dev",
+    # model card: https://huggingface.co/torchao-testing/single-linear-Int4WeightOnlyConfig-v2-0.13-dev
+    "torchao-testing/single-linear-Int4WeightOnlyConfig-v2-0.13-dev",
+    # model card: https://huggingface.co/torchao-testing/single-linear-Int4WeightOnlyConfig-preshuffled-v2-0.13-dev
+    "torchao-testing/single-linear-Int4WeightOnlyConfig-preshuffled-v2-0.13-dev",
+]
+
+
+@unittest.skipIf(not torch.cuda.is_available(), "Need CUDA available")
+@unittest.skipIf(not is_sm_at_least_89(), "Nedd sm89+")
+@unittest.skipIf(
+    is_fbcode(),
+    "Skipping the test in fbcode for now, not sure how to download from transformers",
+)
+class TestLoadAndRunCheckpoint(TestCase):
+    def _test_single_linear_helper(self, model_name):
+        from huggingface_hub import hf_hub_download
+
+        downloaded_model = hf_hub_download(model_name, filename="model.bin")
+        # Load model weights, example inputs and reference output,
+        # run the loaded model and make sure the result matches reference output
+
+        with torch.device("meta"):
+            # 32 and 256 are the args we used when we save the model, see
+            # model card:
+            # https://huggingface.co/torchao-testing/single-linear-FP8-v2-0.13-dev
+            model = torch.nn.Sequential(
+                torch.nn.Linear(32, 256, dtype=torch.bfloat16, device="cuda")
+            )
+        with open(downloaded_model, "rb") as f:
+            model.load_state_dict(torch.load(f), assign=True)
+
+        downloaded_example_inputs = hf_hub_download(
+            model_name, filename="model_inputs.pt"
+        )
+        with open(downloaded_example_inputs, "rb") as f:
+            example_inputs = torch.load(f)
+        downloaded_output = hf_hub_download(model_name, filename="model_output.pt")
+        with open(downloaded_output, "rb") as f:
+            ref_output = torch.load(f)
+
+        output = model(*example_inputs)
+        self.assertTrue(torch.equal(output, ref_output))
+
+    @common_utils.parametrize("model_name", _DEPRECATED_SINGLE_LINEAR_MODEL_NAMES)
+    def test_deprecated_single_linear(self, model_name):
+        self._test_single_linear_helper(model_name)
+
+    @common_utils.parametrize("model_name", _SINGLE_LINEAR_MODEL_NAMES)
+    def test_single_linear(self, model_name):
+        """Test that we can load and run the quantized linear checkpoint with saved sample input
+        and match the saved output, to make sure there is no BC breaking changes
+        when we make changes to tensor subclass implementations
+        """
+        if (
+            "Float8DynamicActivationFloat8WeightConfig" in model_name
+            and not is_sm_at_least_90()
+        ):
+            return unittest.skip("FP8 checkpoint is produced in SM90+")
```

Comment:
should we ensure we have CI with sm90+ running this test?

Reply:
yeah it's tested here: https://github.com/pytorch/ao/blob/083361bc3f7addc505a0f994a923f4ae9f54388e/.github/workflows/1xH100_tests.yml#L50

added this skip because it fails in this test: https://github.com/pytorch/ao/blob/083361bc3f7addc505a0f994a923f4ae9f54388e/.github/workflows/1xL4_tests.yml#L50 (SM89)